window/window.go

```diff
@@ -727,6 +731,28 @@ func (w *window) decrement(count int64) {
 	}
 }
 
+func (w *window) shiftLeft(count int64) {
+	_, bytes, err := w.readBytes(w.cursor, 1)
+	if err != nil {
+		return
+	}
+	w.replace(w.cursor, bytes[0]<<byte(mathutil.MaxInt64(count, 1)%256))
```

Comment:
It seems it works but I can't understand why this line does not result into overflow exception.

Reply:
Roughly speaking, while in Rust, for example, shift operators panic if `right-hand-side >= 8` for `u8` type, in Go, shift operators do not panic for any integer values.

## References
- https://doc.rust-lang.org/reference/expressions/operator-expr.html#overflow
- https://golang.org/ref/spec#Arithmetic_operators

Reply:
Ahh, okay, I understand in Go
```
x, y := byte('a'), byte(100)
fmt.Println(x << y)
```
does not panic while
```
fmt.Println((byte('a') << byte(100)))
```
does not even compile. I got confused because I was using gore.

Reply:
It's due to typed constants and [representability](https://golang.org/ref/spec#Representability)...😥